src/JustSaying/Messaging/MessageSerialization/SystemTextJsonSerializer`1.cs

```diff
@@ -0,0 +1,50 @@
+using System.Text.Json;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging.MessageSerialization;
+
+/// <summary>
+/// Provides serialization and deserialization functionality for messages of type <typeparamref name="T"/> using System.Text.Json.
+/// </summary>
+/// <typeparam name="T">The type of message to be serialized or deserialized. Must inherit from <see cref="Message"/>.</typeparam>
+public sealed class SystemTextJsonMessageBodySerializer<T> : SystemTextJsonMessageBodySerializerBase, IMessageBodySerializer where T: Message
+{
+    private readonly JsonSerializerOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonMessageBodySerializer{T}"/> class with default JSON serializer options.
+    /// </summary>
+    public SystemTextJsonMessageBodySerializer()
+    {
+        _options = DefaultJsonSerializerOptions;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonMessageBodySerializer{T}"/> class with custom JSON serializer options.
+    /// </summary>
+    /// <param name="options">The custom <see cref="JsonSerializerOptions"/> to use for serialization and deserialization.</param>
+    public SystemTextJsonMessageBodySerializer(JsonSerializerOptions options)
+    {
+        _options = options;
+    }
+
+    /// <summary>
+    /// Serializes a message to its JSON string representation.
+    /// </summary>
+    /// <param name="message">The message to serialize.</param>
+    /// <returns>A JSON string representation of the message.</returns>
+    public string Serialize(Message message)
+    {
+        return JsonSerializer.Serialize(message, message.GetType(), _options);
+    }
```

Comment:
Can we make this `Serialize<T>(T message)`? `message.GetType()` won't play friendly for native AoT if we add support for that in the future.

Reply:
I can imagine we may not have generic type in all callers. For AoT we can create and store the JTI<T> if available in the constructor (and throw if not available), and use in the serialize and deserialize methods.